actionsList.ts

```diff
@@ -0,0 +1,68 @@
+interface IToDo {
+    name: string,
+    date: Date,
+    finished: boolean,
+}
+
+class Action implements IToDo {
+    constructor(public name: string, public date: Date, public finished: boolean) {
+    }
+
+    public finishTask() {
+        this.finish();
+    }
+
+    private finish() {
+        this.finished = true;
+    }
+}
+
+class ActionsList {
+    public toDos = [];
+
+    public addToDo(Action: object) {
+        this.toDos.push(Action);
+    }
+
+    public getToDos() {
+        return this.toDos;
+    }
+
+    public listNotFinished() {
+        const notFinished = this.toDos.filter(t => !t.finished);
+        console.table(notFinished);
+    }
+
+    public listAll() {
+        console.table(this.toDos);
+    }
+
+    public removeToDo(action) {
+        if (!action.finished) {
+            for (var i = 0; i < this.toDos.length; i++) {
+                if (this.toDos[i].name === action.name) {
+                    this.toDos.splice(i, 1)
+                }
+            }
+        }
+    }
```

Comment:
Implementarea ta este corecta, dar trebuie sa stii ca se poate simplifica, iar de cele mai multe ori, cautarea unui obiect intr-o lista se face cu o proprietate a obiectului, in cazul tau, name.
```
const list = [{name: 'a'}, {name: 'b'}];
function removeElement(name) {
	const element = list.find(el => el.name === name);
	if (element) {
		list.splice(list.indexOf(element), 1);
	}
}
removeElement('a');
```